pkg/apis/core/validation/cloudprofile.go

```diff
@@ -423,6 +423,10 @@ func validateBastion(spec *core.CloudProfileSpec, fldPath *field.Path) field.Err
 		return allErrs
 	}
 
+	if spec.Bastion.MachineType == nil && spec.Bastion.MachineImage == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, spec.Bastion, "bastion section needs a machine type or machine image"))
+	}
```

Comment:
Does it really only need one of them and not both? I would have thought that you need a machine type and a corresponding operating system image.

Reply:
The logic is currently as following:
- if only the machineType is set, we take the first imageType that supports this arch and from this take the newest supported version
- if only the machineImage is defined, we'll take the machineType with the lowest amount of CPU that matches the arch of the defined image